app/services.js

```diff
@@ -53,229 +31,99 @@ airTrafficControlApp.service('systemBootService', ['$location', '$q', '$timeout'
-            else if (null == aircraftA) {
-                return -1;
-            }
-            else if (null == aircraftB) {
-                return 1;
-            }
-        }
-        
-        // Compare priorities
-        numericalPriorityAircraftA = self.determineNumericalPriority(aircraftA);
-        numericalPriorityAircraftB = self.determineNumericalPriority(aircraftB);
-        
-        if (numericalPriorityAircraftA > numericalPriorityAircraftB) {
-            //aircraft B has a higher priority
-            return -1;
-        }
-        else if (numericalPriorityAircraftA < numericalPriorityAircraftB) {
-            //aircraft A has a higher priority
-            return 1;
-        }
-        
-        // Both have same priority
-        // compare time when the aircrafts were added to system
-        var timeAddedToSystemComparison = self.compareTimeAddedToSystem(aircraftA.timeAdded, aircraftB.timeAdded);
-        
-        if (timeAddedToSystemComparison !== 0) {
-            // If time when added to system is different, we already have our result
-            return timeAddedToSystemComparison;
-        }
-        
-        // Everything is the same, so the comparison returns a 0
-        return 0;
-    };
-    
-    self.compareTimeAddedToSystem = function (aircraftATimeAdded, aircraftBTimeAdded) {
-        // Time added to system comparison based on problem specifications
-        // Note that if the aircraft was added later, it will have lower priority
-        // than any aircraft added earlier
-        if (aircraftATimeAdded === aircraftBTimeAdded) {
-            return 0;
-        }
-        
-        if (aircraftATimeAdded > aircraftBTimeAdded) {
-            return -1;
-        }
-        else{
-            return 1;
-        }
-        
-        // we will never come here
-        return 0;
-    }
-    
-    // Passenger Large = Prio 0
-    // Passenger Small = Prio 1
-    // Cargo Large = Prio 2
-    // Cargo Small = Prio 3
-    self.determineNumericalPriority = function(aircraft){
-        var priority = 0;
-        
-        switch(aircraft.type + "|" + aircraft.size) {
-            case "Passenger|Large":
-                priority = 0;
-                break;
-            case "Passenger|Small":
-                priority = 1;
-                break;
-            case "Cargo|Large":
-                priority = 2;
-                break;
-            case "Cargo|Small":
-                priority = 3;
-                break;
-        }
-        
-        return priority;
-    }
-    
-    self.numberOfAircraftPriorities = function(){
-        return 4;
-    }
-
-}]);
-
-
 
 
 // Queue management service [Enqueue/Dequeue]
 // Use the array to avoid the minification problem
-airTrafficControlApp.service('queueManagementService', ['$q', '$timeout', 'aircraftPriorityDeterminationService', function($q, $timeout, aircraftPriorityDeterminationService) {
+airTrafficControlApp.service('queueManagementService', [function() {

```

Comment:
1) use module pattern
2) please add comments to the top of this function explaining the functionality of this service. It seems overly complicated; the contract per the assignment is that it responds to: boot, enqueue(aircraft) and dequeue();

so three public methods, and possibly a few private ones to handle this; but I am reading a lot more functions for those three simple actions.